src/node_api_async.h

```diff
@@ -0,0 +1,20 @@
+#ifndef SRC_NODE_API_ASYNC_H_
+#define SRC_NODE_API_ASYNC_H_
+
+#include <stdlib.h>
+#include "node_api.h"
+#include "node_api_async_types.h"
+
+EXTERN_C_START
+
+NAPI_EXTERN napi_work napi_create_async_work();
+NAPI_EXTERN void napi_delete_async_work(napi_work w);
+NAPI_EXTERN void napi_async_set_data(napi_work w, void* data);
+NAPI_EXTERN void napi_async_set_execute(napi_work w, void (*execute)(void*));
+NAPI_EXTERN void napi_async_set_complete(napi_work w, void (*complete)(void*));
+NAPI_EXTERN void napi_async_set_destroy(napi_work w, void (*destroy)(void*));
```

Comment:
The reason that the async APIs were included was to help on the abi stability front.  It seems to be a common use case for modules to need these async APIs. Nan provides an AsyncWorker class with similar functionality as does the C++ wrapper that will go along with the N-API, and we needed them for a number of the modules we ported. Without these APIs, if libuv changes any existing modules would break and need to be modified and recompiled.  I think that would be a surprise for those modules using the C++ wrapper as all other APIs used by the wrapper would be protected through the N-API.